### ISSUE TITLE
Dialog not showing up

### ISSUE THREAD

**Issue by cirogg** [Naveentp/In-App-Review-API-Demo#1]
Hi! I am trying to implement the same thing as you and my Review Dialogue is not showing up.

I tried cloning your project and testing it, and the same thing happened.

My applicationID is correct. In my app and in yours happens the same, just go trough the completeListener and nothing show up.

Tried with differents google accounts and in two different phones.

**Comment by jonathancoding** [comment]
I had the same issue. @Naveentp I clicked the surprise button on screen 1 and go back button on screen 2 for multiple times but nothing happened. Do you know what steps are we missing here?

**Comment by swayangjit** [comment]
@Naveentp Same here tried with a valid id available in play store, still unable to see the  popup.

**Comment by Naveentp** [comment]
Hi @cirogg, @jonathancoding, @swayangjit

I have added the README.  There are few things which we need to watchout.

1. Make sure you give applicationId of an app which is available in Play Store.
2. If you have already given rating in the past, Rationg dialog will not show up. Sign-in with different account in play store and try again.
3. Due to quota limitations, calling a launchReviewFlow method might not always display a dialog. It should not be linked with any button click event.
4. While testing one of my apps, I had to create a signed apk to see the Review card but for the other app, debug build worked just fine.

**Comment by swayangjit** [comment]
@Naveentp No luck.. tried creating a signed apk too.

**Comment by cirogg** [comment]
@Naveentp I tried this
1: Do everything in the onCreate and onResume of my MainActivity.
2: Tried with debug apk and Signed APK. The applicationID is a valid one
3: Tried a new project with the valid applicationID.
4: Tried too pulling this project and still no luck.
5: Loggin with another account in the playstore
6: A different phone


I know it is a basic step by step guide to follow but i dont know why it is not working. :(. I get the message of the onComplete and nothing else happens.

**Comment by Naveentp** [comment]
Here are some solutions to [troubleshoot](https://developer.android.com/guide/playcore/in-app-review/test#troubleshooting) specific issues. It looks like people are facing the [same issue](https://issuetracker.google.com/issues/162969017). 

Could you guys try adding your email id to Internal test track for the app in play store and test? If it works for Internal test track activated email id and not for other email ids, it must be a bug or quota issue.

**Comment by jonathancoding** [comment]
“I had to create a signed apk to see the Review card but for the other app, debug build worked just fine.”

Just wondering what's causing that delta？ Are all your apps published in PlayStore？

**Comment by jonathancoding** [comment]
> Hi! I am trying to implement the same thing as you and my Review Dialogue is not showing up.
> 
> I tried cloning your project and testing it, and the same thing happened.
> 
> My applicationID is correct. In my app and in yours happens the same, just go trough the completeListener and nothing show up.
> 
> Tried with differents google accounts and in two different phones.

Do you have to download this demo app from Google play store and then override the apk with AS and then test?

**Comment by mustafiz012** [comment]
> @Naveentp No luck.. tried creating a signed apk too.

Any luck till now?

**Comment by swayangjit** [comment]
> > @Naveentp No luck.. tried creating a signed apk too.
> 
> Any luck till now?

Nope. Better to test it in a testing track internally by uploading it to playstore.

**Comment by MittapalliHareesh** [comment]
Hi, I had written code in onCreate method. And trying on signed build only.   In reviewManager.launchReviewFlow it's going to success and complete methods. But not getting review dialog.

                val flow = reviewManager.launchReviewFlow(this, it)
                flow.addOnSuccessListener {
                    Toast.makeText(this, "SUCCESS", Toast.LENGTH_LONG).show()
                }
                flow.addOnFailureListener {
                    Toast.makeText(this, "FAIL", Toast.LENGTH_LONG).show()
                }
                flow.addOnCompleteListener {
                    Toast.makeText(this, "COMPLETED", Toast.LENGTH_LONG).show()
                }

It's printing  "SUCCESS" and "COMPLETED" but not able to see any review dialog. Can anyone help with this?

**Comment by UtkuGlsvn** [comment]
I have that same problem , I applied the java version of the code in my own app, I tried a different e-mail account and your code in its app, it still did not show dialog

**Comment by MittapalliHareesh** [comment]
Hi, Same issue i had faced,  Raised Google bug. 

Please look on this ticket  https://issuetracker.google.com/issues/166260001  for more details.
